src/core_plugins/kibana/public/visualize/listing/visualize_landing_page_table.js

```diff
@@ -71,17 +45,11 @@ export class VisualizeLandingPageTable extends React.Component {
   }
 
   onFilter = (newFilter) => {
-    const PAGE_SIZE = 3;
+    this.setState(filterItems());
```

Comment:
I really like the names you're suggesting for the actions -- much more understandable!

I do think passing in setState is a step too far in terms of abstraction. This shifts the call site of setState from one place to another, and I think the only effect on someone reading the code is that now it's more difficult to figure out the role of `filter`. By keeping the call site of setState local, we form a relationship between it and `filter`. This relationship helps the reader intuit the role of `filter` (to derive and return a new state object that is somehow filtered).

I'm also hesitant to pass `this.props.fetch` into it.  This would give it knowledge of network requests and asynchronous logic. A lot more complexity than it has now, but for what benefit?